### PR TITLE
FMO-OS CICD initial

### PR DESCRIPTION
**Pull request by jsvapiav**
- Add build action and workflow to make labeled builds from tagged commits
- Add pre-commit simple check - mostly just call nix flake show for now
- Add test fails for PRs try to change any workflows to prevent unauthorized access to sensitive information

### CODE REVIEW

File: .github/actions/build-action/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+echo "::group::Input validation"
+[ ! "$BUILD_TARGET" ] && err_exit 1 "BUILD_TARGET undefined"
+[ ! "$CACHIX_TOKEN" ] && err_exit 1 "CACHIX_TOKEN undefined"
+[ ! "$RA_TOKEN" ] && err_exit 1 "RA_TOKEN undefined"
+echo "::endgroup::"
+
+echo "::group::Install cachix"
+nix-env -iA cachix -f https://cachix.org/api/v1/install
+echo "Using cachix version:"
+cachix --version || err_exit 1 "Cachix intallation has failed. Fail"
```
Comment (JoonasOnatsu) [inline .github/actions/build-action/entrypoint.sh:39]: Small typo, fix 'intallation' to 'installation'.

File: .github/actions/upload-action/upload.sh [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+[ ! "$JFROG_UNAME" ] && err_exit 1 "JFROG_UNAME undefined"
+[ ! "$JFROG_TOKEN" ] && err_exit 1 "JFROG_TOKEN undefined"
+
+for input in $INPUT_PATHS; do
+  SOURCE_DIR=$(echo "$input" | cut -d ":" -f 1)
+  DEST_DIR=$(echo "$input" | cut -d ":" -f 2)
+  echo "SOURCE_DIR=$SOURCE_DIR"
+  echo "DEST_DIR=$DEST_DIR"
+  [ ! "$SOURCE_DIR" ] && err_exit 1 "SOURCE_DIR undefined"
+  [ ! "$DEST_DIR" ] && err_exit 1 "SOURCE_DIR undefined"
```
Comment (JoonasOnatsu) [inline .github/actions/upload-action/upload.sh:29]: Please fix error message.

File: .github/actions/upload-action/upload.sh [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+jf rt ping
+
+for input in $INPUT_PATHS; do
+  SOURCE_DIR=$(echo "$input" | cut -d ":" -f 1)
+  DEST_DIR=$(echo "$input" | cut -d ":" -f 2)
+
+  UPLOAD_DIR=$SOURCE_DIR
+  if [ "$BUILD_NUM" ]; then
+    UPLOAD_DIR=$SOURCE_DIR-b$BUILD_NUM
+    mv "$SOURCE_DIR" "$UPLOAD_DIR"
```
Comment (JoonasOnatsu) [inline .github/actions/upload-action/upload.sh:45]: This move happens only when 'BUILD_NUM' is defined. Should this happen also when it is not defined?

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      - name: Free Disk Space (Ubuntu)
+        uses: jlumbroso/free-disk-space@main
+        with:
+          # this might remove tools that are actually needed,
+          # if set to "true" but frees about 6 GB
+          tool-cache: false
+        
+          # all of these default to true, but feel free to set to
+          # "false" if necessary for your workflow
+          android: true
```
Comment (JoonasOnatsu) [inline .github/workflows/build.yml:39]: Do we need android or dotnet tooling, should these be removed?

**Review by JoonasOnatsu — CHANGES_REQUESTED** [review]
Please review comments and fix.

File: .github/workflows/build.yml (hunk 4 above)
Comment (jsvapiav) [reply .github/workflows/build.yml:39]: here `true` means to be removed, `false` opposite

File: .github/actions/build-action/entrypoint.sh (hunk 1 above)
Comment (jsvapiav) [reply .github/actions/build-action/entrypoint.sh:39]: Cool. Thanks!

File: .github/actions/upload-action/upload.sh (hunk 2 above)
Comment (jsvapiav) [reply .github/actions/upload-action/upload.sh:29]: One more catch! Thanks!

File: .github/actions/upload-action/upload.sh (hunk 3 above)
Comment (jsvapiav) [reply .github/actions/upload-action/upload.sh:45]: No, if it is not defined `UPLOAD_DIR == SOURCE_DIR`

Comment (JoonasOnatsu) [reply .github/actions/upload-action/upload.sh:45]: Ok cool.

File: .github/workflows/build.yml (hunk 4 above)
Comment (JoonasOnatsu) [reply .github/workflows/build.yml:39]: Ah ok my bad

**Review by JoonasOnatsu — APPROVED** [review]

**Pull request merged**